horace_core/configuration/@hor_config/hor_config.m

```diff
@@ -296,14 +296,20 @@
             if use
                 % Configure mex usage
                 % --------------------
-                [~, n_errors, can_combine_with_mex] = check_horace_mex();
+                [~, n_errors, can_combine_with_mex,can_use_custom_mpi] = check_horace_mex();
                 if n_errors>0
                     use = false;
                     warning('HOR_CONFIG:set_use_mex',...
                         ' mex files can not be initiated, Use mex set to false');
                 end
                 if ~can_combine_with_mex
-                    config_store.instance().store_config(obj,'combine_sqw_using','matlab');
+                    config_store.instance().store_config('hpc_config','combine_sqw_using','matlab');
+                end
+                if ~can_use_custom_mpi
+                    pc = parallel_config;
+                    if ismember(pc.parallel_cluster,{'mpiexec_mpi','slurm_mpi'})
```

Comment:
Why if these two are present in `parallel_cluster` would it default to herbert? What is this `if` statement actually checking?

Reply:
because `~can_use_custom_mpi` means that `cpp_communicator` does not work. These two frameworks use `cpp_communicator` as the transport, so they are not avaliable if it does not work.